resources/assets/components/CampaignSingle/index.js

```diff
@@ -24,15 +24,37 @@ class CampaignSingle extends React.Component {
     this.deletePost = this.deletePost.bind(this);
     this.filterPosts = this.filterPosts.bind(this);
     this.getPostsByPaginatedLink = this.getPostsByPaginatedLink.bind(this);
+    this.createTagSlug = this.createTagSlug.bind(this);
+    this.getPostsByTag = this.getPostsByTag.bind(this);
   }
 
   componentDidMount() {
     this.getPostsByStatus('accepted', this.props.campaign.id);
   }
 
-  // Filter posts based on status.
-  filterPosts(status) {
-    this.getPostsByStatus(status.toLowerCase(), this.props.campaign.id);
+  // Filter posts based on status or tag(s).
+  filterPosts(filter) {
+    // If the filter is a status, make API call to get posts by status.
+    if (filter === 'Pending' || filter === 'Accepted' || filter === 'Rejected') {
+      this.getPostsByStatus(filter.toLowerCase(), this.props.campaign.id);
+    } else {
+      // If the filter is a tag, make the API call to get posts by tag.
+      var tagSlug = this.createTagSlug(filter);
+
+      this.getPostsByTag(tagSlug, this.props.campaign.id);
+    }
+
+  }
+
+  // Create tag slug
+  createTagSlug(filter) {
+    var splitFilter = filter.toLowerCase().split(' ');
+
+    if (splitFilter[splitFilter.length-1] === '👻') {
```

Comment:
Hahaha I'd be on board with removing the 👻  from the dropdown in order to not have to do this. But at the same time, I love this line.